addons/popochiu/engine/objects/prop/popochiu_prop.gd

```diff
@@ -137,6 +137,98 @@ func get_total_frames() -> int:
+func queue_play_animation_backwards(name: StringName = &"", custom_blend: float = -1) -> Callable:
+	return func (): await play_animation_backwards(name, custom_blend)
+
+
+## Will Play the Popochiu Prop Animation Player [param name] animation backwards on the Popochiu
+## Prop if the [prop name] animation exits.
+## This method is a shorthand for play_animation() with custom_speed = -1.0 and from_end = true
+func play_animation_backwards(name: StringName = &"", custom_blend: float = -1) -> void:
+	if not has_node("AnimationPlayer"): return
+	$AnimationPlayer.play_backwards(name, custom_blend)
+
+
+## Will stop the animation that is currently playing on the Popochiu Prop.
+## The animation position is reset to 0 and the custom_speed is reset to 1.0.
+## Set [param keep_state] to true to avoid the animation to be updated visually.
+## [i]This method is intended to be used inside a [method Popochiu.queue] of instructions.[/i]
+func queue_stop_animation(keep_state: bool = false) -> Callable:
+	return func (): await stop_animation(keep_state)
+
+
+## Will stop the animation that is currently playing on the Popochiu Prop.
+## The animation position is reset to 0 and the custom_speed is reset to 1.0.
+## Set [param keep_state] to true to avoid the animation to be updated visually.
+func stop_animation(keep_state: bool = false) -> void:
+	if not has_node("AnimationPlayer"): return
+	$AnimationPlayer.stop(keep_state)
+
+## Will pause the animation that is currently playing on the Popochiu Prop.
+## Call play_animation() without any parameter to resume the animation from the paused animation frame.
+## [i]This method is intended to be used inside a [method Popochiu.queue] of instructions.[/i]
+func queue_pause_animation() -> Callable:
+	return func (): await pause_animation()
+
+
+## Will pause the animation that is currently playing on the Popochiu Prop.
+## Call play_animation() without any parameter to resume the animation from the paused animation frame.
+func pause_animation() -> void:
+	if not has_node("AnimationPlayer"): return
+	$AnimationPlayer.pause()
+
+
+## Will return true if an animation is playing, otherwise it will return false.
+func is_animation_playing() -> bool:
+	if not has_node("AnimationPlayer"): return false
+	return $AnimationPlayer.is_playing()
+
+
+## Will return the string name of the currently assigned animation key name on the Popochiu Prop
+## AnimationPlayer
+func get_assigned_animation() -> String:
+	if not has_node("AnimationPlayer"): return ''
+	return $AnimationPlayer.assigned_animation
+
+## Will set the animation key name for the currently assigned animation on the Popochiu Prop
+## AnimationPlayer
+func set_assigned_animation(name: StringName) -> void:
+	if not has_node("AnimationPlayer"): return
+	$AnimationPlayer.assigned_animation = name
+
+
+## Will return the current Popochiu Prop animation position in seconds.
+## returns -1.0 if there is an error.
```

Comment:
```suggestion
## Returns the current animation position (in seconds) in the [AnimationPlayer] node.
## If there is an error, it will return [code]-1.0[/code].
```